core/rawdb/chain_freezer.go

```diff
@@ -114,60 +158,39 @@ func (f *chainFreezer) freeze(db ethdb.KeyValueStore) {
 				return
 			}
 		}
-		// Retrieve the freezing threshold.
-		hash := ReadHeadBlockHash(nfdb)
-		if hash == (common.Hash{}) {
-			log.Debug("Current full block hash unavailable") // new chain, empty database
+		number, err := f.freezeThreshold(nfdb)
```

Comment:
Maybe rename `number` to `threshold`, makes is easier to understand down in the code.